### PR TITLE
fix(pack): elide implicit type imports without breaking classic JSX

### PR DESCRIPTION
**Pull request by fireairforce**
## Summary

- default TypeScript import elision to non-verbatim behavior so imports used only as types do not become runtime resolutions against declaration files
- preserve classic JSX React bindings through the earlier TypeScript transform, including framework facade imports such as @alipay/bigfish/react
- preserve per-file JSX directives until the React transform and remove the temporary marker from final output
- express server-reference graph-discovery transitions as side-effect-only imports so TypeScript elision cannot remove those build graph edges
- add snapshot coverage for extensionless declaration-file imports, classic JSX, automatic JSX import sources, and explicit verbatimModuleSyntax behavior

## Compatibility notes

- unused value imports are now elided by default; modules imported for side effects should use a side-effect-only import or explicitly enable verbatimModuleSyntax
- the classic JSX preserver currently recognizes default/namespace bindings named React; custom JSX pragma aliases remain an edge case to evaluate while this PR is in draft

## Test Plan

- cargo fmt --check
- cargo clippy --all-targets -- -D warnings --no-deps
- tombi format --check
- biome ci
- typos
- cargo test -p pack-tests --test snapshot test_tests__snapshot__basic__server_function -- --exact --nocapture
- cargo test -p pack-tests --test snapshot dts_implicit_type_import -- --nocapture
- cargo test -p pack-tests --test snapshot test_tests__snapshot__basic__jsx -- --exact --nocapture
- cargo test -p pack-tests --test snapshot test_tests__snapshot__library__basic -- --exact --nocapture
- cargo test -p pack-tests --test snapshot test_tests__snapshot__typescript__verbatim_module_syntax -- --exact --nocapture

### CODE REVIEW

File: crates/pack-core/src/shared/transforms/jsx_import_preserver.rs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                        self.react_ident = Some(default_spec.local.clone());
+                    }
+                }
+                // `import * as React from "react"`
+                ImportSpecifier::Namespace(ns_spec) => {
+                    if self.react_ident.is_none() && &*ns_spec.local.sym == "React" {
+                        self.react_ident = Some(ns_spec.local.clone());
+                    }
+                }
+                ImportSpecifier::Named(_) => {}
```
[the remark this answers is not in the document]
Comment (fireairforce) [reply crates/pack-core/src/shared/transforms/jsx_import_preserver.rs]: Fixed in 6727cc793. The preserver now resolves the root bindings referenced by `@jsx` and `@jsxFrag` (including named imports such as `h`/`Fragment`), while continuing to ignore explicit type-only imports. I also added a classic TSX snapshot regression that verifies the custom factory bindings remain and an unrelated `.d.ts` import is still elided.

**Review by elrrrrrrr — APPROVED** [review]
